### ISSUE TITLE
Bug: calcite-input date range picker changes the wrong input

### ISSUE THREAD

**Issue by noahmulfinger** [Esri/calcite-components#2547]
### Actual Behavior

In certain cases, when I am choosing a new value for the first input in the date range picker, it will instead change the second input.

### Expected Behavior

My expectation is that only the first date picker would be updated when I modify the first input. Same applies to the second input.

### Reproduction Steps or Sample

1. Go to https://esri.github.io/calcite-components/?path=/story/components-controls-inputdatepicker--range
1. Click on the first date input and select 12/13 in the date picker. It should correctly update the first input to 12/13 and open the second date picker
1. Refresh the page to reset.
1. Click on the first date input and select 12/14 in the date picker. It will unexpectedly set the *second* input to 12/14 and open the second date picker
1. You can repeat that with any date higher than 12/14 and it will do the same behavior

### Relevant Info

Tested in Firefox and Chrome

_Version_: `@esri/calcite-components@latest`

- [ ] CDN
- [ ] NPM package
- [x] Storybook build

cc @isaacviel

**Comment by julio8a** [comment]
Verified on master, demo pages.

**Comment by noahmulfinger** [comment]
I also just took another look at this (on the Storybook), and its seems partially fixed, but the behavior still seems unexpected to me. When I choose 12/14 as in the steps I listed above, it still updates the second input. I would expect that if I click the first input, any action I take would only update the first date. 

Happy to open a second issue if this requires more discussion of the expected behavior.
